src/Common/src/System/Net/CookieParser.cs

```diff
@@ -543,18 +544,60 @@ internal string GetString()
             return _tokenizer.GetCookieString();
         }
 
-        private static void SetCookieName(Cookie cookie, string name)
+        private static MethodInfo s_internalSetNameMethod;
+        private static MethodInfo InternalSetNameMethod
         {
-            try
+            get
             {
-                cookie.Name = name;
+                if (s_internalSetNameMethod == null)
+                {
+                    // We need to use Cookie.InternalSetName instead of the Cookie.set_Name wrapped in a try catch block, as
+                    // Cookie.set_Name keeps the original name if the string is empty or null.
+                    // Unfortunately this API is internal so we use reflection to access it. The method is cached for performance reasons.
```

Comment:
@Priya91 unfortunately, we can't do
```
try
{
    cookie.Name = name;
}
catch (CookieException)
{
    Debug.Assert(cookie.Name == string.Empty);
}
```
because if `name` is null or empty, `Cookie.set_Name` doesn't override the old name, whereas `Cookie.InternalSetName` does override the old name unconditionally